statsd/worker.go

```diff
@@ -71,6 +73,9 @@ func (w *worker) processMetric(m metric) error {
 }
 
 func (w *worker) shouldSample(rate float64) bool {
+	// rand.NewSource is not thread safe
+	w.randomLock.Lock()
+	defer w.randomLock.Unlock()
```

Comment:
`defer` had an overhead until Go 1.14, since we support older go versions we may want to avoid using it here since it's a hot path

Reply:
details: https://medium.com/i0exception/runtime-overhead-of-using-defer-in-go-7140d5c40e32